src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -691,25 +691,13 @@ private boolean getOrReplaceAllIndices(final Object request, final IndicesProvid
             }
             ((UpdateRequest) request).index(newIndices.length!=1?null:newIndices[0]);
         } else if (request instanceof SingleShardRequest) {
-            final SingleShardRequest<?> gr = (SingleShardRequest<?>) request;
-            final String[] indices = gr.indices();
-            final String index = gr.index();
-
-            final List<String> indicesL = new ArrayList<String>();
-
-            if (index != null) {
-                indicesL.add(index);
-            }
-
-            if (indices != null && indices.length > 0) {
-                indicesL.addAll(Arrays.asList(indices));
-            }
-
-            String[] newIndices = provider.provide(indicesL.toArray(new String[0]), request, true);
-            if (!checkIndices(request, newIndices, true, allowEmptyIndices)) {
+            final SingleShardRequest<?> singleShardRequest = (SingleShardRequest<?>) request;
+            final String index = singleShardRequest.index();
+            String[] indices = provider.provide(index == null ? null : new String[]{index}, request, true);
+            if (!checkIndices(request, indices, true, allowEmptyIndices)) {
                 return false;
             }
-            ((SingleShardRequest) request).index(newIndices.length!=1?null:newIndices[0]);
+            ((SingleShardRequest) request).index(indices.length!=1?null:indices[0]);
```

Comment:
nit: sorry if it was not clear from my previous comment. The cast is not necessary. We can use `singleShardRequest` instead.